core/trino-spi/src/main/java/io/trino/spi/subfield/Subfield.java

```diff
@@ -0,0 +1,337 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.subfield;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+// Class to represent subfield. Direct referenced from Presto
+public class Subfield
```

Comment:
Subfield as in row type or ?

Reply:
We don't need a dedicated abstraction in the SPI to represent this. This information is available (and should be encoded) in the structure of ConnectorExpressions passed to the connector APIs. We can have utilities in the plugin toolkit module to extract the necessary info to make it easier for connector implementers.

Reply:
This is already represented as ArrayFieldDereference in this PR, then in this case we will still need to extract it to a form that readers can use regardless of table formats?